src/Neo/Ledger/Blockchain.cs

```diff
@@ -339,7 +339,11 @@ private VerifyResult OnNewExtensiblePayload(ExtensiblePayload payload)
 
         private VerifyResult OnNewTransaction(Transaction transaction)
         {
-            if (system.ContainsTransaction(transaction.Hash)) return VerifyResult.AlreadyExists;
+            switch (system.ContainsTransaction(transaction.Hash))
+            {
+                case ContainsTransactionType.ExistsInPool: return VerifyResult.AlreadyInPool;
+                case ContainsTransactionType.ExistsInLedger: return VerifyResult.AlreadyExists;
```

Comment:
`TestValidTransaction` should be fixed wrt this change (currently it's failing and it's expected).